File: prisma/seeds/clip.ts
import { PrismaClient } from "@prisma/client";
import { fetchSubCollection } from "./utils";

const prisma = new PrismaClient({ log: ["query", "info", "error", "warn"] });

export const importClips = async () => {
  const clips = await fetchSubCollection("clips");
  for (const clip of clips) {
    await prisma.clip.upsert({
      where: {
        gameId_userId: {
          gameId: clip.game.ref.id,
          userId: clip.profile.ref.id,
        },
      },
      create: {
        gameId: clip.game.ref.id,
        userId: clip.profile.ref.id,
        createdAt: clip.createdAt.toDate(),
      },
      update: {},
    });
  }
};

File: backend/app/models/clip.rb
# frozen_string_literal: true

# ゲームのクリップを表すモデル

# @attr [String] id ID
# @attr [Game] game ゲーム
# @attr [Profile] profile プロフィール
# @attr [Date] created_at 作成日
# @attr [Date] updated_at 更新日
class Clip < ApplicationRecord
  belongs_to :game
  belongs_to :profile
end
